### PR TITLE
fix: config engine fixes (TokenizationSpoke ProxyAdmin owner, role relabeling, position manager action ordering)

### PR DESCRIPTION
**Pull request by avniculae**
This PR contains three config engine fixes.

**TokenizationSpoke ProxyAdmin owner** — `TokenizationSpokeDeployer.deploy` set the ProxyAdmin owner to `msg.sender`, which under governance execution (PayloadsController → executor → delegatecall) resolves to the PayloadsController, so engine-deployed TokenizationSpokes ended up with an owner that can't exercise ownership. `TokenizationSpokeConfig` now has an explicit `proxyAdminOwner` field, threaded through the deployer (zero owner reverts). The tokenization config is validated all-or-nothing: fully unset skips deployment, partially set reverts with `InvalidTokenizationSpokeConfig` instead of being silently ignored. New `GovernanceTopology.t.sol` executes an asset-listing payload through the real call→delegatecall chain and asserts the deployed spoke's ProxyAdmin owner. README and diagram corrected accordingly (`msg.sender` is the PayloadsController and must never be used).

**Role relabeling** — relabeling an already-labeled role through the engine reverted with `AccessManagerRoleAlreadyLabeled`, since `AccessManagerEnumerable` requires clearing the label first. `AccessManagerEngine` now clears before setting. Empty string remains the skip sentinel, so clearing without relabeling still requires a direct `labelRole` call.

**Position manager action ordering** — `AaveV4Payload` ran spoke registrations before role renouncements, so a payload renouncing and deregistering the same spoke reverted (`renouncePositionManagerRole` is `onlyRegisteredSpoke`). Renouncements now run first.

Note: the `TokenizationSpokeConfig` struct change is ABI-breaking for existing `AssetListing` builders.

### CODE REVIEW

File: src/config-engine/README.md [hunk 1]
```diff
@@ -22,17 +22,17 @@ The four groups, and the virtual functions in each, are listed below.
...
-| `hubAssetCapsResets()`        | `AssetCapsReset`        | Reset asset caps                                                                                      |
-| `hubSpokeDeactivations()`     | `SpokeDeactivation`     | Deactivate a Spoke                                                                                    |
-| `hubSpokeCapsResets()`        | `SpokeCapsReset`        | Reset Spoke caps                                                                                      |
+| Function                      | Struct                  | Purpose                                                                                                                                                    |
+| ----------------------------- | ----------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `hubAssetListings()`          | `AssetListing`          | List a new asset on a Hub. Optionally deploys a TokenizationSpoke if `tokenization` is set (`name`, `symbol` and `proxyAdminOwner` are then all required). |
```
Comment (DhairyaSethi) [inline src/config-engine/README.md]: i dont think we need proxy admin owner as a requirement, i would keep it optional, have AaveV4Payload constructor take in defaultOwner (which is executor)

File: tests/helpers/mocks/config-engine/MockGovernanceExecutor.sol [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  error FailedActionExecution();
+
+  constructor(address owner) {
+    OWNER = owner;
+  }
+
+  function executeTransaction(address target, bytes memory data) external returns (bytes memory) {
+    require(msg.sender == OWNER, OnlyOwner());
+    (bool success, bytes memory resultData) = target.delegatecall(data);
+    require(success, FailedActionExecution());
```
Comment (DhairyaSethi) [inline tests/helpers/mocks/config-engine/MockGovernanceExecutor.sol]: add tests with non zero `value` cases also

File: tests/config-engine/GovernanceTopology.t.sol [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+pragma solidity ^0.8.0;
+
+import 'tests/config-engine/BaseConfigEngine.t.sol';
+
+import {ProxyHelper} from 'tests/utils/ProxyHelper.sol';
+import {MockGovernanceExecutor} from 'tests/helpers/mocks/config-engine/MockGovernanceExecutor.sol';
+
+/// @dev Production-style payload: all action data lives in immutables or literals. `execute()`
+/// runs via delegatecall inside the Executor, so payload storage is not readable at execution time.
+contract TokenizationListingPayload is AaveV4Payload {
```
Comment (Kogaroshi) [inline tests/config-engine/GovernanceTopology.t.sol]: let's move this to mocks ?

File: tests/config-engine/HubEngine.t.sol
Comment (Kogaroshi) [inline tests/config-engine/HubEngine.t.sol:815]: can we extend this test to ensure skip was enforced (no new deploy, no new entry in spoke list)

File: src/config-engine/README.md (hunk 1 above)
Comment (DhairyaSethi) [reply src/config-engine/README.md]: actually prob config engine should always set executor as owner, its not fine if its configurable

Comment (avniculae) [reply src/config-engine/README.md]: so it's fine as is, or you'd like to see some change here?

Comment (DhairyaSethi) [reply src/config-engine/README.md]: typo, its NOT fine as is, i dont think it should be configurable

Comment (avniculae) [reply src/config-engine/README.md]: who should be the owner then? we don't have address book here. we can use the owner of the hub proxy admin, but not sure I prefer that over the current approach

Comment (Kogaroshi) [reply src/config-engine/README.md]: I think he means calls to the deploy fct should always pass `address(this)` => the Executor.

Comment (DhairyaSethi) [reply src/config-engine/README.md]: executor, as i said above, either similar similar to v3 which is call on configurator which deploys with owner, or keep it as an immutable on the payload contract which is passed to the engine

Comment (DhairyaSethi) [reply src/config-engine/README.md]: https://github.com/aave-dao/aave-v3-origin/blob/main/src/contracts/extensions/v3-config-engine/AaveV3ConfigEngine.sol#L25-L26

Comment (DhairyaSethi) [reply src/config-engine/README.md]: > I think he means calls to the deploy fct should always pass address(this) => the Executor.

also possible, yes. either approach works

File: tests/config-engine/HubEngine.t.sol
Comment (Kogaroshi) [reply tests/config-engine/HubEngine.t.sol:815]: 96e56a378f6fd343999e48a2a9e1f7aeab6a6285

File: tests/config-engine/GovernanceTopology.t.sol (hunk 3 above)
Comment (Kogaroshi) [reply tests/config-engine/GovernanceTopology.t.sol]: 96e56a378f6fd343999e48a2a9e1f7aeab6a6285

File: tests/helpers/mocks/config-engine/MockGovernanceExecutor.sol (hunk 2 above)
Comment (Kogaroshi) [reply tests/helpers/mocks/config-engine/MockGovernanceExecutor.sol]: 96e56a378f6fd343999e48a2a9e1f7aeab6a6285

File: src/config-engine/README.md (hunk 1 above)
Comment (Kogaroshi) [reply src/config-engine/README.md]: 96e56a378f6fd343999e48a2a9e1f7aeab6a6285

File: src/config-engine/libraries/TokenizationSpokeDeployer.sol [hunk 4]
```diff
@@ -10,20 +10,28 @@ import {TokenizationSpokeInstance} from 'src/spoke/instances/TokenizationSpokeIn
...
 /// using the Safe Singleton Factory.
 library TokenizationSpokeDeployer {
+  /// @dev Thrown when the proxy admin owner is the zero address.
+  error InvalidProxyAdminOwner();
+
   /// @notice Deploys a TokenizationSpokeInstance implementation and TransparentUpgradeableProxy via CREATE2
   /// through the Safe Singleton Factory.
-  /// @dev The proxy admin owner is set to `msg.sender`.
+  /// @dev The proxy admin owner must be passed explicitly, never derived from the execution
+  /// context (e.g. `msg.sender`, which resolves to the PayloadsController under governance).
```
Comment (miguelmtzinf) [inline src/config-engine/libraries/TokenizationSpokeDeployer.sol]:
```suggestion
  /// @dev The proxy admin owner must be passed explicitly, never derived from execution context 
```

File: src/config-engine/README.md [hunk 5]
```diff
@@ -8,7 +8,7 @@ The `AaveV4ConfigEngine` is a helper smart contract to abstract best practices w
 
 Based on experience reviewing governance payloads for Aave V3, the config engine provides a type-safe, composable interface that covers the most common administrative operations: Hub configuration, Spoke configuration, AccessManager role management, and PositionManager administration.
 
-The engine itself is **stateless** — it never stores data of its own. Payloads invoke it via `delegatecall`, so every external call the engine makes executes in the payload's (governance executor's) context and with the executor's permissions.
+The engine itself is **stateless** — it never stores data of its own. Payloads invoke it via `delegatecall`, so every external call the engine makes executes in the governance executor's context (`address(this)`) and with the executor's permissions. See [Execution context](#execution-context) for the full topology — in particular, `msg.sender` is the PayloadsController and must never be used.
```
Comment (miguelmtzinf) [inline src/config-engine/README.md]:
```suggestion
The engine itself is **stateless** — it never stores data of its own. Payloads invoke it via `delegatecall`, so every external call the engine makes executes in the governance executor's context and with the executor's permissions. See [Execution context](#execution-context) for the full topology.
```

File: src/config-engine/AaveV4Payload.sol [hunk 6]
```diff
@@ -404,21 +404,24 @@ abstract contract AaveV4Payload {
   }
 
   /// @notice Executes all Position Manager configuration actions via delegatecall to the engine.
+  /// @dev Role renouncements run before Spoke registrations: renouncing requires the Spoke to
+  /// still be registered on the position manager, so a payload can renounce and deregister the
+  /// same Spoke in one batch.
```
Comment (miguelmtzinf) [inline src/config-engine/AaveV4Payload.sol]:
```suggestion
  /// @dev PositionManager Role renouncements happen before Spoke registrations: renouncing requires the 
  /// Spoke to still be registered on the position manager itself.
```

File: src/config-engine/README.md [hunk 7]
```diff
@@ -100,8 +100,11 @@ When `execute()` is called, actions run in the following fixed order:
    5. Dynamic reserve config updates
    6. Position manager updates
 5. **PositionManager actions** (in order):
-   1. Spoke registrations
-   2. Role renouncements
+   1. Role renouncements
```
Comment (miguelmtzinf) [inline src/config-engine/README.md]:
```suggestion
   1. Spoke PositionManager Role renouncements
```

File: src/config-engine/README.md [hunk 8]
```diff
@@ -100,8 +100,11 @@ When `execute()` is called, actions run in the following fixed order:
...
+   2. Spoke registrations
+
+   Renouncements run first because renouncing requires the Spoke to still be registered on the position manager — this allows renouncing and deregistering the same Spoke in one payload.
```
Comment (miguelmtzinf) [inline src/config-engine/README.md]:
```suggestion
   Spoke PositionManager role renouncements run first because renouncing requires the Spoke to still be registered on the position manager — this allows renouncing and deregistering the same Spoke in one payload.
```

Comment (Kogaroshi) [reply src/config-engine/README.md]: 782c28e60b1144bd41ffc38b5826a3e1f6f4e476

File: src/config-engine/README.md (hunk 7 above)
Comment (Kogaroshi) [reply src/config-engine/README.md]: 782c28e60b1144bd41ffc38b5826a3e1f6f4e476

File: src/config-engine/AaveV4Payload.sol (hunk 6 above)
Comment (Kogaroshi) [reply src/config-engine/AaveV4Payload.sol]: 782c28e60b1144bd41ffc38b5826a3e1f6f4e476

File: src/config-engine/README.md (hunk 5 above)
Comment (Kogaroshi) [reply src/config-engine/README.md]: 782c28e60b1144bd41ffc38b5826a3e1f6f4e476

File: src/config-engine/libraries/TokenizationSpokeDeployer.sol (hunk 4 above)
Comment (Kogaroshi) [reply src/config-engine/libraries/TokenizationSpokeDeployer.sol]: 782c28e60b1144bd41ffc38b5826a3e1f6f4e476

File: tests/helpers/mocks/config-engine/TokenizationListingPayload.sol [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+pragma solidity ^0.8.0;
+
+import {AaveV4Payload} from 'src/config-engine/AaveV4Payload.sol';
+import {IAaveV4ConfigEngine} from 'src/config-engine/interfaces/IAaveV4ConfigEngine.sol';
+import {IHubConfigurator} from 'src/hub/interfaces/IHubConfigurator.sol';
+import {IAssetInterestRateStrategy} from 'src/hub/interfaces/IAssetInterestRateStrategy.sol';
+
+/// @dev Production-style payload: all action data lives in immutables or literals. `execute()`
+/// runs via delegatecall inside the Executor, so payload storage is not readable at execution time.
+contract TokenizationListingPayload is AaveV4Payload {
```
Comment (avniculae) [inline tests/helpers/mocks/config-engine/TokenizationListingPayload.sol]: I don't like that the name does not suggest mock/test. should we adjust?

Comment (Kogaroshi) [reply tests/helpers/mocks/config-engine/TokenizationListingPayload.sol]: 78b0a3eb57c35fa8a1eb36672da88efd0bb19624

File: src/config-engine/interfaces/IAaveV4ConfigEngine.sol [hunk 10]
```diff
@@ -15,11 +15,15 @@ import {IAssetInterestRateStrategy} from 'src/hub/interfaces/IAssetInterestRateS
 /// the universal KEEP_CURRENT sentinel. Boolean fields use uint256 (0=false, 1=true, KEEP_CURRENT=skip).
 interface IAaveV4ConfigEngine {
   /// @notice Parameters for tokenization of an asset on a Hub when listing the asset.
-  /// @dev addCap The add cap for the TokenizationSpoke (0 means no tokenization).
+  /// @dev Tokenization is skipped only when all fields are unset. Otherwise `name`, `symbol` and
+  /// `proxyAdminOwner` are all required; a partially set config reverts.
+  /// @dev addCap The add cap for the TokenizationSpoke.
+  /// @dev proxyAdminOwner The owner to set on the ProxyAdmin of the deployed TokenizationSpoke.
   /// @dev name The name for the TokenizationSpoke.
   /// @dev symbol The symbol for the TokenizationSpoke.
```
Comment (miguelmtzinf) [inline src/config-engine/interfaces/IAaveV4ConfigEngine.sol]: let's clarify, what's the unset value here pls

File: src/config-engine/interfaces/IAaveV4ConfigEngine.sol [hunk 11]
```diff
@@ -304,7 +308,8 @@ interface IAaveV4ConfigEngine {
   /// @dev admin The new admin role identifier (KEEP_CURRENT_UINT64 to skip).
   /// @dev guardian The new guardian role identifier (KEEP_CURRENT_UINT64 to skip).
   /// @dev grantDelay The new grant delay (KEEP_CURRENT_UINT32 to skip).
-  /// @dev label The label string (empty string to skip).
+  /// @dev label The label string (empty string to skip). An already-labeled role is cleared before
+  /// relabeling. Clearing a label without setting a new one requires a direct `labelRole` call.
```
Comment (miguelmtzinf) [inline src/config-engine/interfaces/IAaveV4ConfigEngine.sol]: this clarification seems to belong to somewhere else

File: src/config-engine/README.md (hunk 1 above)
Comment (miguelmtzinf) [inline src/config-engine/README.md]:
```suggestion
| `hubAssetListings()`          | `AssetListing`          | List a new asset on a Hub. Optionally deploys a TokenizationSpoke if all tokenization params are set (`name`, `symbol` and `proxyAdminOwner`). |
```

Comment (Kogaroshi) [reply src/config-engine/README.md]: 3abcb3c48fb4730e625da148ed2d74b588e1e112

File: src/config-engine/interfaces/IAaveV4ConfigEngine.sol (hunk 10 above)
Comment (Kogaroshi) [reply src/config-engine/interfaces/IAaveV4ConfigEngine.sol]: 3abcb3c48fb4730e625da148ed2d74b588e1e112

File: src/config-engine/interfaces/IAaveV4ConfigEngine.sol (hunk 11 above)
Comment (Kogaroshi) [reply src/config-engine/interfaces/IAaveV4ConfigEngine.sol]: 3abcb3c48fb4730e625da148ed2d74b588e1e112

File: src/config-engine/README.md [hunk 12]
```diff
@@ -142,6 +145,15 @@ Several engine functions inspect which fields differ from `KEEP_CURRENT` and cho
...
-- All sub-engine code runs in the **payload's storage and `msg.sender` context** (i.e. the governance executor).
 - Neither the config engine nor the sub-engines hold any storage, permissions, or admin keys.
 - All HubConfigurator, SpokeConfigurator, AccessManager, and PositionManager calls originate from the governance executor's address.
+
+### Execution context
+
+In production the payload itself executes via delegatecall: the PayloadsController **calls** `Executor.executeTransaction`, which **delegatecalls** `payload.execute()`. Since `msg.sender` is preserved across delegatecall, for all engine code:
+
+- `address(this)` is the **Executor** — this is the identity holding permissions, and the address external calls originate from.
+- `msg.sender` is the **PayloadsController** — it must never be used, for ownership, permissions, or anything else. Deriving an owner from `msg.sender` is how the Paxos deployment shipped TokenizationSpoke ProxyAdmins owned by the PayloadsController; any address a deployment or configuration needs (e.g. `TokenizationSpokeConfig.proxyAdminOwner`) must be passed explicitly in the action structs.
```
Comment (miguelmtzinf) [inline src/config-engine/README.md]:
```suggestion
- `msg.sender` is the **PayloadsController**. It must never be used, for ownership, permissions, or anything else. Deriving an owner from `msg.sender` is bad practice: any address a deployment or configuration needs (e.g. `TokenizationSpokeConfig.proxyAdminOwner`) must be passed explicitly in the action structs.
```

File: src/config-engine/README.md [hunk 13]
```diff
@@ -142,6 +145,15 @@ Several engine functions inspect which fields differ from `KEEP_CURRENT` and cho
...
 
-- All sub-engine code runs in the **payload's storage and `msg.sender` context** (i.e. the governance executor).
 - Neither the config engine nor the sub-engines hold any storage, permissions, or admin keys.
 - All HubConfigurator, SpokeConfigurator, AccessManager, and PositionManager calls originate from the governance executor's address.
+
+### Execution context
+
+In production the payload itself executes via delegatecall: the PayloadsController **calls** `Executor.executeTransaction`, which **delegatecalls** `payload.execute()`. Since `msg.sender` is preserved across delegatecall, for all engine code:
+
+- `address(this)` is the **Executor** — this is the identity holding permissions, and the address external calls originate from.
```
Comment (miguelmtzinf) [inline src/config-engine/README.md]:
```suggestion
- `address(this)` is the **Executor**. It is the identity holding permissions, and the address external calls originate from.
```

File: src/config-engine/README.md [hunk 14]
```diff
@@ -142,6 +145,15 @@ Several engine functions inspect which fields differ from `KEEP_CURRENT` and cho
...
+- `msg.sender` is the **PayloadsController** — it must never be used, for ownership, permissions, or anything else. Deriving an owner from `msg.sender` is how the Paxos deployment shipped TokenizationSpoke ProxyAdmins owned by the PayloadsController; any address a deployment or configuration needs (e.g. `TokenizationSpokeConfig.proxyAdminOwner`) must be passed explicitly in the action structs.
+- Payload storage is not readable during execution — action data must live in immutables, constants, or literals returned by the overridden virtual functions.
```
Comment (miguelmtzinf) [inline src/config-engine/README.md]:
```suggestion
- Payload storage is not readable during execution: action data must live in immutables, constants, or literals returned by the overridden virtual functions.
```

File: tests/config-engine/AaveV4Payload.t.sol [hunk 15]
```diff
@@ -1021,11 +1021,67 @@ contract AaveV4PayloadTest is BaseConfigEngineTest {
...
+    emit ISpoke.SetUserPositionManager(USER, address(freshPm), false);
+    payload.execute();
+
+    // the position manager is still active on the Spoke, so isPositionManager being false
+    // proves the approval itself was cleared
+    assertTrue(spoke1().isPositionManagerActive(address(freshPm)));
+    assertFalse(spoke1().isPositionManager(USER, address(freshPm)));
+    assertFalse(freshPm.isSpokeRegistered(address(spoke1())));
+  }
+
```
Comment (miguelmtzinf) [inline tests/config-engine/AaveV4Payload.t.sol:1085]: we can add a test for the peculiar scenario where we register spoke in the positionmanager and renounce. This will revert

File: tests/config-engine/GovernanceTopology.t.sol [hunk 16]
```diff
@@ -0,0 +1,86 @@
...
+import 'tests/config-engine/BaseConfigEngine.t.sol';
+
+import {ProxyHelper} from 'tests/utils/ProxyHelper.sol';
+import {MockGovernanceExecutor} from 'tests/helpers/mocks/config-engine/MockGovernanceExecutor.sol';
+import {MockTokenizationListingPayload} from 'tests/helpers/mocks/config-engine/MockTokenizationListingPayload.sol';
+
+/// @dev Executes engine payloads through the real governance topology
+/// (PayloadsController → Executor → delegatecall payload → delegatecall engine), where
+/// `msg.sender` is the PayloadsController and `address(this)` is the Executor.
+contract ConfigEngineGovernanceTopologyTest is BaseConfigEngineTest {
```
Comment (miguelmtzinf) [inline tests/config-engine/GovernanceTopology.t.sol:14]: can we clarify the purpose of this test file? we wanna ensure initialized permissions not affected by `msg.sender` and `address(this)` essentially?

File: src/config-engine/interfaces/IAaveV4ConfigEngine.sol [hunk 17]
```diff
@@ -15,11 +15,15 @@ import {IAssetInterestRateStrategy} from 'src/hub/interfaces/IAssetInterestRateS
...
   /// @notice Parameters for tokenization of an asset on a Hub when listing the asset.
-  /// @dev addCap The add cap for the TokenizationSpoke (0 means no tokenization).
-  /// @dev name The name for the TokenizationSpoke.
-  /// @dev symbol The symbol for the TokenizationSpoke.
+  /// @dev Tokenization is skipped only when all fields are unset. Otherwise `name`, `symbol` and
+  /// `proxyAdminOwner` are all required; a partially set config reverts.
+  /// @dev addCap The add cap for the TokenizationSpoke, unset value : 0.
+  /// @dev proxyAdminOwner The owner to set on the ProxyAdmin of the deployed TokenizationSpoke, unset value : address(0).
+  /// @dev name The name for the TokenizationSpoke, unset value : ''.
+  /// @dev symbol The symbol for the TokenizationSpoke, unset value : ''.
```
Comment (miguelmtzinf) [inline src/config-engine/interfaces/IAaveV4ConfigEngine.sol]:
```suggestion
  /// @dev addCap The add cap for the TokenizationSpoke.
  /// @dev proxyAdminOwner The owner to set on the ProxyAdmin of the deployed TokenizationSpoke (address(0) when unset).
  /// @dev name The name for the TokenizationSpoke ('' when unset).
  /// @dev symbol The symbol for the TokenizationSpoke ('' when unset).
```

`addCap` can be 0 anyway, no change there

File: tests/config-engine/GovernanceTopology.t.sol (hunk 16 above)
Comment (Kogaroshi) [reply tests/config-engine/GovernanceTopology.t.sol:14]: Yes, it's meant to run test with the same flow of calls & delegatecalls than during a real Governance Paylaod execution

File: src/config-engine/interfaces/IAaveV4ConfigEngine.sol (hunk 17 above)
Comment (Kogaroshi) [reply src/config-engine/interfaces/IAaveV4ConfigEngine.sol]: ffceac91647a60c11b9423222cb58ca07fca9e6d

File: tests/config-engine/AaveV4Payload.t.sol (hunk 15 above)
Comment (Kogaroshi) [reply tests/config-engine/AaveV4Payload.t.sol:1085]: ffceac91647a60c11b9423222cb58ca07fca9e6d

File: src/config-engine/README.md (hunk 14 above)
Comment (Kogaroshi) [reply src/config-engine/README.md]: ffceac91647a60c11b9423222cb58ca07fca9e6d

File: src/config-engine/README.md (hunk 13 above)
Comment (Kogaroshi) [reply src/config-engine/README.md]: ffceac91647a60c11b9423222cb58ca07fca9e6d

File: src/config-engine/README.md (hunk 12 above)
Comment (Kogaroshi) [reply src/config-engine/README.md]: ffceac91647a60c11b9423222cb58ca07fca9e6d

File: src/config-engine/libraries/HubEngine.sol [hunk 18]
```diff
@@ -19,8 +19,12 @@ library HubEngine {
   /// KEEP_CURRENT sentinel. All fields must be explicitly set when the strategy changes.
   error InvalidIrDataWithNewStrategy();
 
+  /// @dev Thrown when a tokenization config is partially set. Either all fields are unset (no
+  /// TokenizationSpoke) or `name`, `symbol` and `proxyAdminOwner` must all be provided.
+  error InvalidTokenizationSpokeConfig();
+
   /// @notice Lists new assets on Hubs via the HubConfigurator.
-  /// @dev When `tokenization.name` & `tokenization.symbol` are defined, also deploys a TokenizationSpoke (impl + proxy) via
+  /// @dev When `tokenization` is set, also deploys a TokenizationSpoke (impl + proxy) via
```
Comment (miguelmtzinf) [inline src/config-engine/libraries/HubEngine.sol]:
```suggestion
  /// @dev When tokenization data is set, also deploys a TokenizationSpoke (impl + proxy) via
```

File: tests/config-engine/GovernanceTopology.t.sol (hunk 16 above)
Comment (miguelmtzinf) [reply tests/config-engine/GovernanceTopology.t.sol:14]: let's clarify with a comment pls

File: src/config-engine/libraries/HubEngine.sol (hunk 18 above)
Comment (Kogaroshi) [reply src/config-engine/libraries/HubEngine.sol]: 371b28b7ee066f11b7db0cdd856243c8c3f96d7d

File: tests/config-engine/GovernanceTopology.t.sol (hunk 16 above)
Comment (Kogaroshi) [reply tests/config-engine/GovernanceTopology.t.sol:14]: ce986dcc0bc0b881409b3817884b0e80a7a25ffa

File: src/config-engine/libraries/AccessManagerEngine.sol [hunk 19]
```diff
@@ -32,6 +33,8 @@ library AccessManagerEngine {
   }
 
   /// @notice Updates role configuration (admin, guardian, grant delay, label) via AccessManager.
+  /// @dev An already-labeled role is cleared before relabeling, as required by the
+  /// AccessManagerEnumerable label tracking.
```
Comment (DhairyaSethi) [inline src/config-engine/libraries/AccessManagerEngine.sol]: don't like the fact this is silently handled here

**Review by DhairyaSethi — DISMISSED** [review]

File: src/config-engine/libraries/AccessManagerEngine.sol (hunk 19 above)
Comment (miguelmtzinf) [reply src/config-engine/libraries/AccessManagerEngine.sol]: we can add a bool to `updates` item, to signal the "replacement" action

File: tests/config-engine/GovernanceTopology.t.sol (hunk 16 above)
Comment (miguelmtzinf) [reply tests/config-engine/GovernanceTopology.t.sol:14]: but this is a test to validate the execution flow does not alter permissions config right? This won't be updated unless we find more usecases like tokenizationSpoke, right?
The title must be smt like "Tests"/"Validates" instead of "Executes"

File: src/config-engine/libraries/AccessManagerEngine.sol (hunk 19 above)
Comment (Kogaroshi) [reply src/config-engine/libraries/AccessManagerEngine.sol]: c47f49c1858afe4ccb8b3e852584358fb7d8ebad

File: tests/config-engine/GovernanceTopology.t.sol (hunk 16 above)
Comment (Kogaroshi) [reply tests/config-engine/GovernanceTopology.t.sol:14]: c47f49c1858afe4ccb8b3e852584358fb7d8ebad

**Review by miguelmtzinf — APPROVED** [review]
